From 81d1ae883db2c5e8350cd2035699c1b319e2c626 Mon Sep 17 00:00:00 2001
From: Yu Ke <ke.yu@intel.com>
Date: Fri, 28 Oct 2011 14:40:51 +0800
Subject: [PATCH] add webmin recipe

commit e48f61d2f6df32a518bcb84db8c6eacfe5435c32

NFS export: fix syntax issue

fix two syntax issue in /etc/exports:
1. if it is exported to everyone, should use "*", e.g. "/export  *(xxx)"
2. explicitly specify subtree_check or no_subtree_check, required by latest nfs-utils

Upstream-Status: Pending

Signed-off-by: Yu Ke <ke.yu@intel.com>

---
 exports/save_export.cgi | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/exports/save_export.cgi b/exports/save_export.cgi
index bcd4f88..0aa64b1 100755
--- a/exports/save_export.cgi
+++ b/exports/save_export.cgi
@@ -50,7 +50,7 @@ else {
 			&error(&text('save_enetmask', $in{'netmask'}));
 		$exp{'host'} = $in{'network'}."/".$in{'netmask'};
 		}
-	elsif ($in{'mode'} == 3) { $exp{'host'} = ""; }
+	elsif ($in{'mode'} == 3) { $exp{'host'} = "*"; }
 	else {
 		$in{'host'} =~ /\*/ || &to_ipaddress($in{'host'}) ||
 			&error(&text('save_ehost', $in{'host'}));
@@ -87,6 +87,11 @@ else {
 	delete($opts{'no_subtree_check'});
 	delete($opts{'subtree_check'});
 	$opts{'no_subtree_check'} = "" if ($in{'no_subtree_check'});
+	if ($in{'no_subtree_check'}) {
+		$opts{'no_subtree_check'} = "";
+	} else {
+		$opts{'subtree_check'} = "";
+	}
 
 	delete($opts{'nohide'});
 	delete($opts{'hide'});
